exporter/loadbalancingexporter/exporter.go

```diff
@@ -189,13 +189,23 @@ func (e *exporterImp) ConsumeTraces(ctx context.Context, td pdata.Traces) error
 	e.updateLock.RLock()
 	defer e.updateLock.RUnlock()
 
+	var wg sync.WaitGroup
+	var errorsLock sync.Mutex
 	var errors []error
 	batches := batchpertrace.Split(td)
 	for _, batch := range batches {
-		if err := e.consumeTrace(ctx, batch); err != nil {
-			errors = append(errors, err)
-		}
+		batch := batch
+		wg.Add(1)
+		go func() {
```

Comment:
That's a great idea! Instead of having a go routine for every batch, we should probably use a worker pool.

Reply:
Every exporter has already a "sending_queue" + "retry", if you use the "sending_queue" you don't need this because that already adds object to a queue and process them async.

Reply:
Yes, that's what I also thought about, but without queue, the processing is sequential. In my opinion, if queue is enabled, then there is no difference, but if disabled then this makes sence

Reply:
In this case here, when using a "groupbytrace" processor before this, there would be only one single go routine here (one trace per batch). When not using that processor, there might be multiple traces per batch, and they are going to be split based on the single batch. So, "sending_queue" wouldn't be effective here either, unless I'm missing something.

Reply:
Once you get the correct Batch, you will call the TraceExporter instance, which usually (I would say always) has it's own SendingQueue, so in the `consumeTrace` where you call `exp.ConsumeTraces(ctx, td)` that call finishes immediately because the element is added to the queue. No point in making this async for such a short operation to perform.

Reply:
By the way, is groupbytrace really efficient before this exporter? In many cases one trace is split among multiple services on different nodes. Given this situation I see no point in using groupbytrace in agent but not in collector

Reply:
> Once you get the correct Batch, you will call the TraceExporter instance

That's absolutely right! I was thinking about "before" this exporter, but "after" this exporter is also valid :-)

> By the way, is groupbytrace really efficient before this exporter? In many cases one trace is split among multiple services on different nodes. Given this situation I see no point in using groupbytrace in agent but not in collector

This should be listed in the readme, but let me know if it isn't or if it's not clear there: basically, I foresee one instance of the otelcol as "load balancer", which would probably sit between an agent and the cluster of collectors. For most cases, the list of backend servers isn't changing that often, and when it changes, only a small spectrum of the trace ID is shuffled to other instances. But for some other cases, the list of backend servers is highly elastic, so that it could make sense to send as many spans from the same trace at the same time as possible, to avoid landing on a different collector.

A future iteration of this exporter could perhaps add some coordination among nodes, so that the loadbalancer itself can be scaled out. The biggest problem with that is ensuring that all nodes have the same cluster view at all times, which would require inter-node communication with some consensus algorithm.

Reply:
> That's absolutely right! I was thinking about "before" this exporter, but "after" this exporter is also valid :-)

I think we should not use the queue_retry logic "before" this exporter, no point. We should enforce that every exporter that we delegate to has the queue_retry enabled and we don't need the current logic.

Reply:
I wouldn't force, but recommend (and perhaps activate it in the `DefaultConfiguration`).

Reply:
@bogdandrutu I see a point, where queue_retry logic should be "before" this exporter. Sending a trace to queued exporter leaks it when the backend hosts are changed. With dns resolver some backends are allowed to disappear, which leads to the loss of every trace in the queue. Having retry logic "before" exporter allows to change endpoint on retries. This approach loses no traces if appropriate queue size is configured

Reply:
I think we are now in the realm of #1690.